sphinx/source/Tutorials/Receiving Messages.rst

```diff
@@ -0,0 +1,196 @@
+     }
+   }
+
+   int main()
+   {
+    isobus::NAME myNAME(0); // Create an empty NAME
+    std::shared_ptr<isobus::InternalControlFunction> myECU = nullptr; // A pointer to hold our InternalControlFunction
+    std::shared_ptr<isobus::PartneredControlFunction> myPartner = nullptr; // A pointer to hold a partner
+
+    // Define a NAME filter for our partner
+    std::vector<isobus::NAMEFilter> myPartnerFilter;
+    const isobus::NAMEFilter virtualTerminalFilter(isobus::NAME::NAMEParameters::FunctionCode, static_cast<std::uint8_t>(isobus::NAME::Function::VirtualTerminal));
+    myPartnerFilter.push_back(virtualTerminalFilter);
+
+    // Set up the hardware layer to use "can0"
+    CANHardwareInterface::set_number_of_can_channels(1);
+    CANHardwareInterface::assign_can_channel_frame_handler(0, "can0");
+    CANHardwareInterface::start();
+
+    // Handle control+c
+    std::signal(SIGINT, signal_handler);
+
+    CANHardwareInterface::add_can_lib_update_callback(update_CAN_network, nullptr);
+    CANHardwareInterface::add_raw_can_message_rx_callback(raw_can_glue, nullptr);
+
+    // Set up NAME fields
+    myNAME.set_arbitrary_address_capable(true);
+    myNAME.set_industry_group(1);
+    myNAME.set_device_class(0);
+    myNAME.set_function_code(static_cast<std::uint8_t>(isobus::NAME::Function::SteeringControl));
+    myNAME.set_identity_number(2);
+    myNAME.set_ecu_instance(0);
+    myNAME.set_function_instance(0);
+    myNAME.set_device_class_instance(0);
+    myNAME.set_manufacturer_code(64);
+
+    // Create our InternalControlFunction
+    myECU = std::make_shared<isobus::InternalControlFunction>(myNAME, 0x1C, 0);
+
+    // Register to receive broadcast PROPA messages
+    isobus::CANNetworkManager::CANNetwork.add_global_parameter_group_number_callback(0xEF00, propa_callback, nullptr);
+
+    // Create our PartneredControlFunction
+    myPartner = std::make_shared<isobus::PartneredControlFunction>(0, myPartnerFilter);
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+
+    std::array<std::uint8_t, isobus::CAN_DATA_LENGTH> messageData = {0}; // Data is just all zeros
+
+    // Send a message to the broadcast address
+    isobus::CANNetworkManager::CANNetwork.send_can_message(0xEF00, messageData.data(), isobus::CAN_DATA_LENGTH, myECU.get());
+
+    // Send a message to our partner (if it is present)
+    isobus::CANNetworkManager::CANNetwork.send_can_message(0xEF00, messageData.data(), isobus::CAN_DATA_LENGTH, myECU.get(), myPartner.get());
+
+    while (true)
+    {
+     // CAN stack runs in other threads. Do nothing forever.
+     std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+    }
+
+    // Clean up the threads
+    CANHardwareInterface::stop();
+
+    return 0;
+   }
+
+This will be tricky to test, as you would need another valid ISO 11783 device on the bus *that has properly address claimed* and is also sending this message in order for the stack to recieve it and pass it to your callback.
```

Comment:
```suggestion
This will be tricky to test, as you would need another valid ISO 11783 device on the bus *that has properly address claimed* and is also sending this message in order for the stack to receive it and pass it to your callback.
```